src/Esprima/Ast/ArrayExpression.cs

```diff
@@ -4,9 +4,9 @@ namespace Esprima.Ast
 {
     public class ArrayExpression : Node, Expression
     {
-        public readonly List<ArrayExpressionElement> Elements;
+        public readonly NodeList<ArrayExpressionElement> Elements;
 
-        public ArrayExpression(List<ArrayExpressionElement> elements) :
+        public ArrayExpression(NodeList<ArrayExpressionElement> elements) :
```

Comment:
Why comment on a deleted line?

> not it's a copy unless a ref, do people need to construct manually

I can't quite parse this. Could you rephrase and/or expand, please?

Reply:
I think the concern here is that when the `ArrayExpression` is constructed, the `elements` collection is passed by value, cloning the whole array.

Reply:
tried to fix a typo, but Sebastien pretty much coined it. I think the struct is copied, even though the referenced array might be intact and happily in heap, only pointer with the struct.

Reply:
Ah, I see the point being raised here.

> cloning the whole array

It doesn't clone the whole array. It (_logically_) copies the structure, which is the _array reference_ plus the count. However, this situation is no different with `NodeList<>` than it was with `List<>` previously. Micro-optimisation concerns should be raised as a separate issue and addressed in a separate PR to avoid bundling too many things in one review. Don't you think?